src/service/document-click.js

```diff
@@ -129,6 +135,43 @@ export class ClickHandler {
     }
   }
 
+  /**
+   * Navigates a window to a URL.
+   *
+   * If opt_requestedBy matches a feature name in a <meta> tag with attribute
+   * name="amp-to-amp-navigation", then treats the URL as an AMP URL (A2A).
+   *
+   * @param {!Window} win
+   * @param {string} url
+   * @param {string=} opt_requestedBy
+   */
+  navigateTo(win, url, opt_requestedBy) {
+    if (!isProtocolValid(url)) {
+      user().error(TAG, 'Cannot navigate to invalid protocol: ' + url);
+      return;
+    }
+
+    // If this redirect was requested by a feature that opted into A2A,
+    // try to ask the viewer to navigate this AMP URL.
+    if (opt_requestedBy) {
+      if (!this.a2aFeatures_) {
+        const meta = this.rootNode_.querySelector(
+            'meta[name="amp-to-amp-navigation"]');
+        this.a2aFeatures_ = (meta && meta.hasAttribute('content'))
+          ? meta.getAttribute('content').split(',').map(s => s.trim())
+          : [];
+      }
+      if (this.a2aFeatures_.indexOf(opt_requestedBy) >= 0) {
+        if (this.viewer_.navigateToAmpUrl(url, opt_requestedBy)) {
+          return;
+        }
+      }
+    }
+
+    // Otherwise, perform normal behavior of navigating the top frame.
+    win.top.location.href = url;
```

Comment:
Why doesn't this handle custom protocols, too?

Reply:
Please make an issue for it.

Reply:
#13385